src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int DEFAULT_SIZE = 0;
+    private static final int SIMPLE_NUMBER_FOR_HASHCODE = 17;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Object[] table = new Object[DEFAULT_INITIAL_CAPACITY];
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
-
+        checkTableSize();
+        addNodeToTable(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int keyHashCode = getHashCode(key);
+        int indexOfBucket = Math.abs(findBucket(keyHashCode));
+        if (table[indexOfBucket] != null) {
+            Node<K, V> currentNode = (Node<K, V>) table[indexOfBucket];
+            while (true) {
+                if (Objects.equals(currentNode.key, key)) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next != null ? currentNode.next : null;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkTableSize() {
+        int threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+        if (size > threshold) {
+            resizeTable();
+        }
+    }
+
+    private void resizeTable() {
+        Object[] newTable = new Object[table.length * 2];
+        Object[] oldTable = table;
+        table = newTable;
+        size = DEFAULT_SIZE;
+        copyNodeFromOldTable(oldTable);
+    }
+
+    private void copyNodeFromOldTable(Object[] oldTable) {
+        for (Object nodeFromOldTable : oldTable) {
+            if (nodeFromOldTable != null) {
+                Node<K, V> currentNode = (Node<K, V>) nodeFromOldTable;
+                while (true) {
+                    put(currentNode.key, currentNode.value);
+                    if (currentNode.next == null) {
+                        break;
+                    }
+                    currentNode = currentNode.next;
+                }
+            }
+        }
+    }
+
+    private void addNodeToTable(K key, V value) {
+        Node<K, V> newNode = new Node<>(getHashCode(key), key, value, null);
+        int indexOfBucket = Math.abs(findBucket(newNode.hash));
```

Comment:
It seems to me that if you used this method somewhere else, it could give a minus index, because you don't have Math.abs in it. But everything works for you so👍